daft/expressions/expressions.py

```diff
@@ -1326,6 +1326,22 @@ def repeat(self, n: int | Expression) -> Expression:
         n_expr = Expression._to_expression(n)
         return Expression._from_pyexpr(self._expr.utf8_repeat(n_expr._expr))
 
+    def substr(self, start: int | Expression, length: int | Expression) -> Expression:
```

Comment:
I am kind of torn between `length: int | Expression = 0` and `length: int | Expression | None = None` as a default case.
Any ideas on which will be easier to handle downstream in Rust code?

Reply:
I don't think it will be that different in difficulty. 

However `length=None` is probably going to be used more often than `length=0`, and is more consistent with [Ibis](https://ibis-project.org/reference/expression-strings#ibis.expr.types.strings.StringValue.substr) and [polars](https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.str.slice.html#polars-expr-str-slice) so I would recommend that!